### ISSUE TITLE
Client API version should be configurable

### ISSUE THREAD

**Issue by dleehr** [Duke-GCB/docker-pipeline#1]
Our cluster nodes are running docker 1.3.2, which is API version 1.15. The current version of docker-py uses API version 1.17.

docker-py can speak version 1.15 but it must be specified as a parameter to Client():

`client = Client(version='1.15',**kwargs_from_env(assert_hostname=False))`

This should be configurable.

**Comment by dansomers** [comment]
Hi Dan - feel free to check docker version on xxxx to see if it is compatible.

**Comment by dleehr** [comment]
Thanks for the tip. That host is running docker 1.4.1, which uses API version 1.17. I took the hostnames and details out of the issue since this is a public github repo.

I was able to get docker working on the cluster by specifying version 1.15 as I mentioned, but hard-coding this version number would be pretty limiting - it would then break on newer hosts. It's unfortunate that docker-py won't auto-negotiate the API version, so this needs to be configurable or tweaked
